### PR TITLE
client: add context to DB.Run()

### PR DESCRIPTION
**Pull request by RaduBerinde**
The main driver for this is to plumb contexts for the timeseries requests.

I am adding ctx to all interfaces in DB (Get, Put, Inc, etc) but that will be a separate change - it will need some coordination around merging to `develop`.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9260)

### CODE REVIEW

File: internal/client/db.go [hunk 1]
```diff
@@ -486,56 +489,55 @@ func (db *DB) prepareToSend(ba *roachpb.BatchRequest) *roachpb.Error {
...
-	Run(b *Batch) error
-}
-
-func runOneResult(r Runner, b *Batch) (Result, error) {
-	if err := r.Run(b); err != nil {
+// getOneErr returns the error for a single-request Batch that was run.
+// runErr is the error returned by Run, b is the Batch that was passed to Run.
+func getOneErr(runErr error, b *Batch) error {
+	if runErr != nil && len(b.Results) > 0 {
+		return b.Results[0].Err
```
Comment (tschottdorf) [inline internal/client/db.go]: Why this over runErr which is also not nil? Is this code movement?

**Comment by tschottdorf** [comment]
LGTM

File: internal/client/db.go (hunk 1 above)
Comment (RaduBerinde) [inline internal/client/db.go]: This is what `runOneResult` (now `getOneReesult`) does..  I just added another function that's more convenient (in the majority of cases we just need the error). I am guessing the error for the request is more precise than the batch error.

**Comment by RaduBerinde** [comment]
@andreimatei Had to fix some conflicts with #8681. Nothing major but in case you want to take a look.

**Comment by RaduBerinde** [comment]
Oops, disregard that, posted in the wrong PR.

**Pull request merged**